### PR TITLE
Multi-crate 2

### PR DESCRIPTION
**Pull request by calebmer**
This is the extension of #9 I briefly touched about in the comments. This structure also allows us to split out more functionality into crates if we want.

A note, not all of these crates have to be published to http://crates.io. It might be easier to understand to split out some things into their own folders even if they just get referenced by path. It would be nice if `graphql-language` was published to http://crates.io separately however.

### CODE REVIEW

**Comment by svmnotn** [comment]
what happened to the rustfmt.toml?

**Comment by calebmer** [comment]
Swapped it with a `.editorconfig` because it looked like the `rustfmt.toml` only was defining tab spaces. If that hurts anyones dev environments, let me know.

Personally I prefer linters over formatters.

**Comment by cksac** [comment]
I would like to have rustfmt do the formatting as gofmt do.  I agree with rustfmt defaut style except tab space width.  That's why I overwrite in rustfmt.toml

File: graphql/Cargo.toml [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 readme = "README.md"
 repository = "https://github.com/cksac/graphql-rs"
 homepage = "https://github.com/cksac/graphql-rs"
-description = """
-A implementation of GraphQL for Rust
-"""
-keywords = ["graphql"]
+
+[lib]
+name = "graphql"
```
Comment (cksac) [inline graphql/Cargo.toml]: Does this mean we declare dep `graphql-rs = "*"` but import use `extern crate graphql`?
I would keep it as `graphql_rs` when import, as user don't need to remember such exception.
User can do `extern crate graphql_rs as graphql;` if wanted.

**Comment by calebmer** [comment]
@cksac 👍

**Pull request merged**